js/ui/diagram/ui.diagram.js

```diff
@@ -152,6 +154,13 @@ class Diagram extends Widget {
         });
     }
 
+    _onPanelPointerUp({ event }) {
+        const preventRefocusSelector = ".dx-textbox";
+        if(!$(event.target).closest(preventRefocusSelector).length) {
```

Comment:
I'd move such condition inside panel's code because diagram widget should not know about implementation of its inner  parts. Also, I suggest to extract the selector const out of method to the beginning of a file (near other constants)